File: src/main/java/p00/Problem_0016.java
package p00;

import java.util.ArrayList;

public class Problem_0016 {

	public static int solve() {

		ArrayList<Integer> number = new ArrayList<>();
		number.add(1);

		for (int i = 0; i < 40; i++) {

			int help = 0;

			for (int h = 0; h < number.size(); h++) {

				int product = number.get(h) * 33554432 + help;
				help = product / 10;

				number.set(h, product - help * 10);

			}

			String string = String.valueOf(help);
			for (int h = string.length() - 1; h >= 0; h--)
				number.add(Integer.valueOf("" + string.charAt(h)));

		}

		int solution = 0;

		for (Integer integer : number) solution += integer;

		return solution;
	}

}

File: src/main/kotlin/p00/Problem0016.kt
package p00

import java.util.*

class Problem0016 {
    fun solve(): Int {
        val number = ArrayList<Int>()
        number.add(1)
        for (i in 0..39) {
            var help = 0
            for (h in number.indices) {
                val product = number[h] * 33554432 + help
                help = product / 10
                number[h] = product - help * 10
            }
            val string = help.toString()
            for (h in string.length - 1 downTo 0) number.add(Integer.valueOf("" + string[h]))
        }
        var solution = 0
        for (integer in number) solution += integer
        return solution
    }
}